From 41ef70f2a1e14b8a49d2d34ad637d72fcf3a95e1 Mon Sep 17 00:00:00 2001
From: Richard Zhu <r65037@freescale.com>
Date: Mon, 18 Apr 2011 18:25:48 +0800
Subject: ENGR00142266-7 eSDHC:Add the GPIO CD/WP on MX53 EVK boards

arm: enable the eSDHC CD/WP on MX53 EVK board

Signed-off-by: Richard Zhu <r65037@freescale.com>
---
 arch/arm/mach-mx5/board-mx53_evk.c |   14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-mx5/board-mx53_evk.c b/arch/arm/mach-mx5/board-mx53_evk.c
index 12f2cd1..5a747fc 100755
--- a/arch/arm/mach-mx5/board-mx53_evk.c
+++ b/arch/arm/mach-mx5/board-mx53_evk.c
@@ -564,6 +564,16 @@ static struct mxc_bus_freq_platform_data evk_bus_freq_data = {
 	.lp_reg_id = "SW2",
 };
 
+static const struct esdhc_platform_data mx53_evk_sd1_data __initconst = {
+	.cd_gpio = EVK_SD1_CD,
+	.wp_gpio = EVK_SD1_WP,
+};
+
+static const struct esdhc_platform_data mx53_evk_sd3_data __initconst = {
+	.cd_gpio = EVK_SD3_CD,
+	.wp_gpio = EVK_SD3_WP,
+};
+
 static int __initdata enable_spdif = { 0 };
 static int __init spdif_setup(char *__unused)
 {
@@ -736,8 +746,8 @@ static void __init mx53_evk_board_init(void)
 	i2c_register_board_info(1, mxc_i2c1_board_info,
 				ARRAY_SIZE(mxc_i2c1_board_info));
 
-	imx53_add_sdhci_esdhc_imx(0, NULL);
-	imx53_add_sdhci_esdhc_imx(1, NULL);
+	imx53_add_sdhci_esdhc_imx(0, &mx53_evk_sd1_data);
+	imx53_add_sdhci_esdhc_imx(2, &mx53_evk_sd3_data);
 
 	spi_register_board_info(mx53_evk_spi_board_info,
 		ARRAY_SIZE(mx53_evk_spi_board_info));
-- 
1.7.9.5

